packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -129,28 +82,28 @@ exports.buildResolvableSet = ({
   defaultLocale,
 }) => {
   const resolvable = new Set()
-  existingNodes.forEach(n => {
-    if (n.contentful_id) {
-      if (process.env.EXPERIMENTAL_CONTENTFUL_SKIP_NORMALIZE_IDS) {
-        resolvable.add(n.contentful_id)
-      } else {
-        // We need to add only root level resolvable (assets and entries)
-        // derived nodes (markdown or JSON) will be recreated if needed.
-        // We also need to apply `fixId` as some objects will have ids
-        // prefixed with `c` and fixIds will recursively apply that
-        // and resolvable ids need to match that.
-        resolvable.add(`${fixId(n.contentful_id)}___${n.sys.type}`)
-      }
+  existingNodes.forEach(node => {
+    if (node.internal.owner === `gatsby-source-contentful`) {
+      // We need to add only root level resolvable (assets and entries)
+      // Derived nodes (markdown or JSON) will be recreated if needed.
+      const key = `${node.contentful_id}___${
+        node.sys.linkType || node.sys.type
+      }`
+      resolvable.add(key)
     }
   })
 
   entryList.forEach(entries => {
     entries.forEach(entry => {
-      resolvable.add(`${entry.sys.id}___${entry.sys.type}`)
+      resolvable.add(
+        `${entry.sys.id}___${entry.sys.linkType || entry.sys.type}`
+      )
     })
   })
   assets.forEach(assetItem =>
-    resolvable.add(`${assetItem.sys.id}___${assetItem.sys.type}`)
+    resolvable.add(
+      `${assetItem.sys.id}___${assetItem.sys.linkType || assetItem.sys.type}`
+    )
```

Comment:
I also applied it to these two, although it did not make a difference in the tests. @axe312ger is this a redundant change or good to keep?

Reply:
See comment above. We should not have links in here at the first place.

It might be that the test fixtures are wrong/outdated.